### PR TITLE
Update readme to include go-githubauth for Application auth

### PR DESCRIPTION
**Pull request by jferrl**
Fixes: #3178.

This PR includes some README changes to include references of a new pkg to authenticate application and installation requests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +140,44 @@ func main() {
...
+func main() {
+ privateKey := []byte(os.Getenv("GITHUB_APP_PRIVATE_KEY"))
+
+ appTokenSource, err := githubauth.NewApplicationTokenSource("app-id", privateKey)
+ if err != nil {
+  fmt.Println("Error creating application token source:", err)
+  return
+ }
+
+ installationTokenSource := githubauth.NewInstallationTokenSource(20, appTokenSource)
```
Comment (gmlewis) [inline README.md:170]: According to this: https://github.com/jferrl/go-githubauth/blob/main/auth.go#L52
the first argument is `id string` so the `20` here looks weird.

Comment (jferrl) [reply README.md]: I've updated the code to be aligned with go-github ids representation. Now both ids are defined as **int64**. Also changed the readme go code examples.

Comment (gmlewis) [reply README.md]: Just FYI - your README.md still needs to be updated - not a biggie - just wanted to let you know.  😄 
https://github.com/jferrl/go-githubauth/blob/main/README.md?plain=1#L48-L51

Comment (jferrl) [reply README.md]: Sure! Im working on it! thanks!

**Review by gmlewis — APPROVED** [review]
Thank you, @jferrl !
LGTM.
Merging.

**Pull request merged**